src/zarr.cpp

```diff
@@ -362,9 +373,17 @@ zarr::Zarr::get(StorageProperties* props) const
     storage_properties_destroy(props);
 
     const std::string dataset_root = dataset_root_.string();
-    const char* filename =
-      dataset_root.empty() ? nullptr : dataset_root.c_str();
-    const size_t bytes_of_filename = filename ? dataset_root.size() + 1 : 0;
+    char* uri = nullptr;
```

Comment:
Is there a reason for this to be a char* (instead of a std::string)?

Reply:
`storage_properties_init` wants a `char*` and I committed to the pointer too early. This is fixed.